### PR TITLE
Fixing User Log Out Logging Message

### PR DESCRIPTION
**Pull request by ccaprio97**
## What’s changing?

Fixing User Log Out Logging Message

## Why?
1989

## This PR has:

- [x] **JIRA ticket(s)** (_Must be listed in PR request and branch name should follow convention NAC-123-description_)<description>

- [ ] **Security Related Changes** (_Security Tab questions have been answered on JIRA ticket(s)_)

- [x] **Commented source code** (_required on public classes and methods, and elsewhere as appropriate_)

- [ ] **Documented API changes** (_required on public classes and methods, and elsewhere as appropriate_)
    - [ ] External API documentation - OpenApi docs and Generated Docs (e.g. https://github.com/18F/piipan/tree/dev/docs/openapi) 
    - [ ] Internal API documentation (e.g. https://github.com/18F/piipan/tree/dev/metrics/docs/openapi)   

- [ ] **Developer documentation** (_for new build/test/API changes or complex portions of the system_)

- [ ] **Automated unit tests** (_to maintain or increase level of code coverage_)

- [ ] **Changes to IAC** (_add any deployment steps that require manual intervention to the draft release notes_)
    - [ ] Have IaC scripts run successfully against an existing deployed environment?
    - [ ] Have IaC scripts run successfully against a newly deployed environment?
    - [ ] Have any new bash scripts been given the appropriate permissions? (e.g. git update-index --chmod=+x $script)
    - [ ] Do the IaC changes require deleting existing resources?

## Anything else?

_Add other details that are helpful for review here_

## Code Review Checklist:

- [ ] Automated checks all pass? (code builds successfully, tests pass, test coverage meets/exceeds threshold, etc.) 
- [ ] Have all requirements and Acceptance Criteria been met?
- **If there are code changes**
    - [ ] Is there no unnecessary complexity? Code is readable, maintainable, etc.
    - [ ] Do the changes follow SOLID principles?
    - [ ] Are the changes robust? Concurrency considered? Performance? Security (e.g. no sql injections)?
    - [ ] Are newly added dependencies needed and appropriate? Are dependency files (e.g. packages.json & packages.lock.json) updated?
    - [ ] For any new projects/solutions, have they been added to code coverage files?
    - [ ] Public classes & methods are documented    
- **If there are API changes**
    - [ ] Does API make sense? Useful and concise?
    - [ ] Are existing API interfaces/contracts intact? i.e. no breaking changes
    - [ ] Are API tests included?
    - [ ] Are API responses appropriate and documented? (200s, 201, 404s, etc.)
    - [ ] Are internals kept private and not leaked into the API?
    - [ ] External API documentation - OpenApi docs and Generated Docs 
    - [ ] Internal API documentation    
- **If there are IAC changes**
    - [ ] IAC Scripts are readable & maintainable
    - [ ] IAC Scripts are documented
    - [ ] IAC scripts pass required checks
    - [ ] IAC scripts were run successfully by author
- **General Logging Questions**
    - [ ] Do new capabilities (or changed capabilities) include logging?
    - [ ] Is PII not logged?
- **General Error Handling Questions**
    - [ ] Has exception handling been added for new capabilities (or changed capabilities)?
    - [ ] Are errors/exceptions properly communicated to end-users? Were there requirements for user facing error messages?
    - [ ] Are errors/exceptions logged?
    - [ ] Is PII kept out of error/exception logging?
- **If there are Developer Documentation Changes**
    - [ ] General Project documentation (under piipan\docs. e.g. iac.md, architecture.md, etc.)
    - [ ] Github documentation additions/changes (Readmes)
- **General automated test questions**
    - [ ] End-to-End Tests provided (if code implements new user story)
    - [ ] Are tests sufficient?
    - [ ] Corner cases considered?
    - [ ] Are performance tests needed?

### CODE REVIEW

File: shared/tests/Piipan.Shared.Tests/Logging/AuthenticationLoggingMiddlewareTests.cs [hunk 1]
```diff
@@ -307,7 +313,7 @@ public async void InvokeAsync_LogOutHeader()
             logger.Verify(m => m.Log(
                 It.Is<LogLevel>(l => l == LogLevel.Information),
                 It.IsAny<EventId>(),
-                    It.Is<It.IsAnyType>((object v, Type _) => v.ToString().Contains($"User logged out: email_claim_type")),
+                    It.Is<It.IsAnyType>((object v, Type _) => v.ToString().Contains($"User logged out: \u0001\0\0\0")),
```
Comment (dborgen2) [inline shared/tests/Piipan.Shared.Tests/Logging/AuthenticationLoggingMiddlewareTests.cs:316]: This seems a bit weird (combined with the code above). Looks like both USER_EMAIL and CLAIMS_LOGGED_KEY are reusing "val". Should this one be outputting "email_claim_type"?

File: shared/src/Piipan.Shared/Logging/AuthenticationLoggingMiddleware.cs [hunk 2]
```diff
@@ -31,16 +32,21 @@ public AuthenticationLoggingMiddleware(RequestDelegate next)
...
                 logger.LogInformation($"User logged out: {loggedInUsersName}");
                 context.Session.SetInt32(CLAIMS_LOGGED_KEY, 0);
             }
             else
             {
                 context.Session.SetInt32(CLAIMS_LOGGED_KEY, 1);
+                string loggedInUsersName = claimsProvider.GetEmail(context.User);
+                if (loggedInUsersName != null)
+                {
+                    context.Session.SetString(USER_EMAIL_KEY, loggedInUsersName);
```
Comment (dborgen2) [inline shared/src/Piipan.Shared/Logging/AuthenticationLoggingMiddleware.cs:45]: Minor Nitpick: If we're not coming from the logout page and we don't have an email, I don't know if we want to be keeping the last email that was used for this session. It might make more sense to clear it out. I can't think of a definitive case, but something like: A user logs out (and also logs out of eauth), and then signs into eauth with a different user, and then goes back to the sign out page, will this keep the previous user's email?

File: shared/tests/Piipan.Shared.Tests/Logging/AuthenticationLoggingMiddlewareTests.cs (hunk 1 above)
Comment (ccaprio97) [reply shared/tests/Piipan.Shared.Tests/Logging/AuthenticationLoggingMiddlewareTests.cs:316]: so I didnt know how to mock and have the get session string ("user email") return the "email_claim_type" and when i tried the build was failing

File: shared/src/Piipan.Shared/Logging/AuthenticationLoggingMiddleware.cs (hunk 2 above)
Comment (ccaprio97) [reply shared/src/Piipan.Shared/Logging/AuthenticationLoggingMiddleware.cs:45]: i think that would hit the new session then wouldnt it? so do you think we should set the email to null on logout?

Comment (dborgen2) [reply shared/src/Piipan.Shared/Logging/AuthenticationLoggingMiddleware.cs:45]: No, I'm simply saying we can probably remove the if statement on line 43. If for some reason the email is null we should probably clear it rather than use the old signed-in email. But it probably doesn't matter too much, I was just thinking it might be possible to have an issue there

Comment (ccaprio97) [reply shared/src/Piipan.Shared/Logging/AuthenticationLoggingMiddleware.cs:45]: Discussed on call and determined it is a non issue

**Review by dborgen2 — APPROVED** [review]
👍
